ipatests/test_integration/test_automember.py

```diff
@@ -0,0 +1,366 @@
+            self.master.run_command(cmd, raiseonerr=False)
+
+    def automember_hostgroup_add(self, hostgroup):
+        """
+        addition of hostgroup/automember entry
+        """
+        cmdlist = [['ipa', 'hostgroup-add', '--desc=%s' % hostgroup,
+                   hostgroup],
+                   ['ipa', 'automember-add', '--type=hostgroup', hostgroup],
+                   ['ipa', 'automember-add-condition',
+                    '--key=nshostlocation', '--type=hostgroup',
+                    '--inclusive-regex=%s' % hostgroup, hostgroup]]
+
+        for cmd in cmdlist:
+            self.master.run_command(cmd)
+
+    def host_automember_setup(self, hostgroup, host):
+        """
+        addition of host entry
+        """
+        self.master.run_command(['ipa', 'host-add', host, '--location=%s'
+                                 % hostgroup, '--force'])
+
+    def host_automember_mod(self, hostgroup, host):
+        """
+        modfication of host entry
+        """
+        self.master.run_command(['ipa', 'host-mod', host, '--location=%s'
+                                 % hostgroup])
+
+    def host_automember_cleanup(self, host):
+        """
+        modfication of host entry
+        """
+        self.master.run_command(['ipa', 'host-del', host])
+
+    def automember_hostgroup_cleanup(self, hostgroup):
+        """
+        cleanup of automember/hostgroup entry
+        """
+        cmdlist = [['ipa', 'automember-del', '--type=hostgroup', hostgroup],
+                   ['ipa', 'hostgroup-del', hostgroup]]
+
+        for cmd in cmdlist:
+            self.master.run_command(cmd, raiseonerr=False)
+
+    def test_modify_user_entry_unmembership(self):
+        # https://pagure.io/389-ds-base/issue/50077
+        """
+        Test that an update on a user entry triggers a re-evaluation of the
+        user's automember groups when auto-unmembership is active.
+        """
+        try:
+            tasks.kinit_admin(self.master)
+            master = self.master
+            search_cmd = [
+                'ldapsearch', '-x',
+                '-D', 'cn=directory manager',
+                '-w', master.config.dirman_password,
+                '-b', 'cn=Auto Membership Plugin,cn=plugins,cn=config',
+                '-s', 'base']
+            result = self.master.run_command(
+                search_cmd,
+                raiseonerr=False
+            )
+            assert result.returncode == 0
+            assert "automemberprocessmodifyops: on" in result.stdout_text
+
+            # testcase setup
+            group1 = 'DallasUsers'
+            group2 = 'HoustonUsers'
+            user1 = 'user1'
+            first = 'first'
+            last = 'user'
+            self.automember_group_add(group1)
+            self.automember_group_add(group2)
+            self.user_automember_setup(group1, user1, first, last)
+
+            result = self.master.run_command(
+                ['ipa', 'group-show', group1]
```

Comment:
This snippet is repeated 18 times throughout the test suite. It is better to avoid repeating by adding a method like
def is_user_member_of_group(self, user, group):
  .....

Then this function can be invoked like
assert is_user_member_of_grpup(user, group)

or assert not is_user_member_of_grpup(user, group)